File: ej2.js
const prompt = require('prompt-sync')();

class Arreglo {
    constructor() {
        this.nombres = [];
    }

    agregarNombre(nombre) {
        this.nombres.push(nombre);
    }

    obtenerNombre(posicion) {
        if (posicion >= 0 && posicion < this.nombres.length)
            return this.nombres[posicion];
        else
            return "Posición inválida";
    }

    cantidad() {
        return this.nombres.length;
    }
}

const arr = new Arreglo();
let continuar;

do {
    let nombre = prompt("Ingrese su nombre: ");
    arr.agregarNombre(nombre);

    let ultimaPos = arr.cantidad() - 1;
    console.log(`El nombre guardado en la posición ${ultimaPos} es: ${arr.obtenerNombre(ultimaPos)}`);

    continuar = prompt("¿Desea continuar? (S/N): ").toUpperCase();
} while (continuar === "S");

File: ej2.java
import java.util.ArrayList;
import java.util.Scanner;

class Arreglo {
    private ArrayList<String> nombres = new ArrayList<>();

    public void agregarNombre(String nombre) {
        nombres.add(nombre);
    }

    public String obtenerNombre(int posicion) {
        if (posicion >= 0 && posicion < nombres.size())
            return nombres.get(posicion);
        else
            return "Posición inválida";
    }

    public int cantidad() {
        return nombres.size();
    }
}

public class ej2 {
    public static void main(String[] args) {
        Arreglo arr = new Arreglo();
        Scanner sc = new Scanner(System.in);
        String nombre;
        String continuar;

        do {
            System.out.print("Ingrese su nombre: ");
            nombre = sc.nextLine();
            arr.agregarNombre(nombre);

            int ultimaPos = arr.cantidad() - 1;
            System.out.println("El nombre guardado en la posición " + ultimaPos + " es: " + arr.obtenerNombre(ultimaPos));

            System.out.print("Desea continuar? (S/N): ");
            continuar = sc.nextLine().toUpperCase();

        } while (continuar.equals("S"));

        sc.close();
    }
}